okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkHttpClients.java

```diff
@@ -221,7 +221,8 @@ private static RemotingOkHttpClient createInternal(
 
         return new RemotingOkHttpClient(
                 client.build(),
-                () -> new ExponentialBackoff(config.maxNumRetries(), config.backoffSlotSize(), new Random()),
+                () -> new ExponentialBackoff(
+                        config.maxNumRetries(), config.backoffSlotSize(), ThreadLocalRandom.current()),
```

Comment:
I use ThreadLocalRandom here because Random will cause requests that start in the same millisecond to back off in the exact same way, which is undesirable.

Reply:
if you want those guarantees it seems marginally safer to have ExponentialBackoff take a `Supplier<Random>` instead of a concrete `Random` (admittedly I haven't yet traced the RemotingOkHttpClient code to check that it makes an ExponentialBackoff per thread)

Reply:
it makes an ExponentialBackoff per call

Reply:
basically it just needs something stable and random - threadlocalrandom is perfect here.